.github/workflows/ci.yml

```diff
@@ -17,28 +17,27 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
       with:
-        python-version: ${{ matrix.python-version }}
+        enable-cache: true
+        # Install a specific version of uv.
+        version: "0.4.x"
+    - name: Set up Python ${{ matrix.python-version }}
+      run: uv python install ${{ matrix.python-version }}
     - name: Install apt dependencies
       run: |
         sudo apt-get update && sudo apt-get -y install plantuml libxml2-dev libxslt-dev python3-dev
-    - name: Install pip & test dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install --upgrade setuptools
-        pip install pytest pre-commit
     - name: Install dependencies for Python
       run: |
-        pip install -c requirements.txt -e .[tests,sphinx,svgsupport,aafiguresupport,mathsupport,rawhtmlsupport]
+        uv sync --all-extras --python ${{ matrix.python-version }}
     - name: Lint
       run: |
-        pre-commit run --all-files --show-diff-on-failure
-      if: matrix.python-version == '3.8'
+        uvx pre-commit run --all-files --show-diff-on-failure
+      if: matrix.python-version == '3.9'
```

Comment:
The intention is to ensure that the pre-commit run only happens once as there's no benefit running it on all python versions. We should fix the matrix definitions properly or bump this to 3.12.

Reply:
Yeah that definitely makes sense. In the current state it was never evaluating to true and running because the matrix no longer had 3.8 (only went from 3.9 -> 3.13).

Of those two options, would you prefer me to put 3.8 into the matrix or bump the conditional to look for 3.12?

Note 3.8 is already in EOL status: https://devguide.python.org/versions/

Reply:
3.12, please. With 3.8 end of life, we aren't interested in supporting it any more and probably need to update our PyPI metadata on the next release too, thinking about it.

Reply:
No worries, updated this line to 3.12.